lisa/util/shell.py

```diff
@@ -340,6 +355,20 @@ def spawn(
                 # Except CommandInitializationError then use minimal shell type.
                 if not have_tried_minimal_type:
                     self._inner_shell._spur._shell_type = spur.ssh.ShellTypes.minimal
+
+                    # Dynamically override that object's method. Here,
+                    # we don't enclose every shell token under single
+                    # quotes anymore. That's an assumption from spur
+                    # that minimal shells will still be POSIX
+                    # compliant--not true for some cases for LISA
+                    # users.
+                    func_type = type(spur.ssh.ShellTypes.minimal.generate_run_command)
```

Comment:
@lubaihua33 Please check if there is regression on distro, which you tested with minimal shell type.